src/frequenz/sdk/actor/power_distributing/_component_status.py

```diff
@@ -0,0 +1,109 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Classes to track the status of components in the microgrid."""
+
+
+import enum
+import typing
+from abc import ABC, abstractmethod
+from collections import abc
+from dataclasses import dataclass
+
+from frequenz.channels import Receiver, Sender
+
+
+@dataclass
+class ComponentPoolStatus:
+    """Status of all components of a certain category in the microgrid."""
+
+    working: set[int]
+    """Set of working component ids."""
+
+    uncertain: set[int]
+    """Set of components to be used only when there are none known to be working."""
```

Comment:
Also probably `collections.abc.Set` so the user can't mutate it either.

Reply:
there's a lot of tiny issues with the code base, the PR is just about renaming some things and modularizing some things.  I did improve some things, but don't want to get too deep into fixing other style issues at the moment in internal code.

Reply:
I tried to do this also, but it doesn't work because we don't have some methods like `intersection`, etc. on the abstract types.

Reply:
Really? That's lame :scream: